src/connectors/hits/__tests__/connectHits-test.ts

```diff
@@ -189,33 +221,45 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/hits/js/#co
       transformItems: items => items.map(() => ({ name: 'transformed' })),
     });
 
-    const helper = algoliasearchHelper({}, '', {});
+    const searchClient = createSearchClient();
+    const helper = algoliasearchHelper(searchClient, '', {});
     helper.search = jest.fn();
 
-    widget.init({
-      helper,
-      state: helper.state,
-      createURL: () => '#',
-    });
+    widget.init!(
+      createInitOptions({
+        helper,
+        state: helper.state,
+      })
+    );
 
     expect(renderFn).toHaveBeenNthCalledWith(
       1,
       expect.objectContaining({ hits: [], results: undefined }),
       expect.anything()
     );
 
-    const hits = [{ name: 'name 1' }, { name: 'name 2' }];
+    const hits = [
+      { objectID: '1', name: 'name 1' },
+      { objectID: '2', name: 'name 2' },
+    ];
 
-    const results = new SearchResults(helper.state, [{ hits }]);
-    widget.render({
-      results,
-      state: helper.state,
-      helper,
-      createURL: () => '#',
-    });
+    const results = new SearchResults(helper.state, [
+      createSingleSearchResponse({ hits }),
+    ]);
+    widget.render!(
+      createRenderOptions({
+        results,
+        state: helper.state,
+        helper,
+      })
+    );
 
-    const expectedHits = [{ name: 'transformed' }, { name: 'transformed' }];
-    expectedHits.__escaped = true;
+    const expectedHits = escapeHits([
+      // @ts-ignore
```

Comment:
We can add `objectID`s to remove the need for `@ts-ignore`.

Reply:
@francoischalifour My transformer function intentionally removes the objectID from the expected hits.